### DISCUSSION TITLE
if x-show is false deselect option

### DISCUSSION THREAD

**Discussion by basvandertogt** [alpinejs/alpine#2708] [1. Help]
The options within the select below are shown or not based on some conditions. When the condition is not met the option disapears. However when this option is already selected it should be deselected. Otherwise this option will be submitted which is incorrect. What is the best approach for this?

```
<select
    x-ref="field"
    x-on:change="field = event.target.options[event.target.selectedIndex].text"
    name="{{ $name }}"
    placeholder="{{ $placeholder }}">

    @forelse($options as $option)
        <option @if($option->filters)x-show="{!! $option->filters !!}"@endif value="{{ $option->id }}" @if($isSelected($option->id)) selected="selected" @endif>
            {{ $option->name }}
        </option>
    @empty
        {!! $slot !!}
    @endforelse
</select>
```

**Comment by SimoTod** [comment]
I assume it's a Livewire question since it looks like it's rendered and updated server-side.
Livewire has hooks, you can find the appropriate one and trigger a change event on your dropdown which will update the value of field

**Reply by basvandertogt** [reply]
The visibility is done by Alpine, not Livewire. I want to de-select the option when it gets hidden by Alpine. Livewire does not know the visibility because it's clientside thing.
